ci-operator/step-registry/mirror-images/enable-qe-catalogsource/mirror-images-enable-qe-catalogsource-commands.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# private mirror registry host
+# <public_dns>:<port>
+MIRROR_REGISTRY_HOST=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
+if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
+    echo "File ${SHARED_DIR}/mirror_registry_url does not exist."
+    exit 1
+fi
+echo "MIRROR_REGISTRY_HOST: ${MIRROR_REGISTRY_HOST}"
+
+# the proxy registry port 6001 for quay.io
+MIRROR_PROXY_REGISTRY_QUAY=`echo "${MIRROR_REGISTRY_HOST}" | sed 's/5000/6001/g' `
+echo "MIRROR_PROXY_REGISTRY_QUAY: ${MIRROR_PROXY_REGISTRY_QUAY}"
+
+# the proxy registry port 6002 for redhat.io
+MIRROR_PROXY_REGISTRY=`echo "${MIRROR_REGISTRY_HOST}" | sed 's/5000/6002/g' `
+echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        # cat "${SHARED_DIR}/proxy-conf.sh"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+# set the registry auths for the cluster
+function set_cluster_auth () {
+    # get the registry configures of the cluster
+    echo "$ oc extract secret/pull-secret -n openshift-config --confirm --to /tmp "
+    if oc extract secret/pull-secret -n openshift-config --confirm --to /tmp ; then 
+        # reminder: there is no brew pull secret here
+        # echo "$ cat /tmp/.dockerconfigjson"
+        # cat /tmp/.dockerconfigjson
+        # add the custom registry auth to the .dockerconfigjson of the cluster
+        registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+        jq --argjson a "{\"${MIRROR_PROXY_REGISTRY_QUAY}\": {\"auth\": \"$registry_cred\"}, \"${MIRROR_PROXY_REGISTRY}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "/tmp/.dockerconfigjson" > /tmp/new-dockerconfigjson
+        # echo "$ cat /tmp/new-dockerconfigjson"
+        # cat /tmp/new-dockerconfigjson
+        # set the registry auth for the cluster
+        if oc set data secret/pull-secret -n openshift-config  --from-file=.dockerconfigjson=/tmp/new-dockerconfigjson ; then
+            echo "set the mirror registry auth succeessfully."
+        else
+            echo "!!! fail to set the mirror registry auth"
+            exit 1
```

Comment:
As a function, suggest to use `return` instead of `exit`

Reply:
Fixed it, thanks!